integration-tests/smoke/ccip/ccip_test.go

```diff
@@ -206,3 +208,797 @@ func TestTokenTransfer(t *testing.T) {
+		require.NoError(t, err)
+		require.GreaterOrEqual(t, srctokenBalance.Int64(), twoCoins.Int64())
+
+		// Assign Src to the Home Chain Selector and destChain to the Feed Chain Selector
+
+		// get the header for the destination chain and the relevant block number
+		latesthdr, err := destChain.Client.HeaderByNumber(testcontext.Get(t), nil)
+		require.NoError(t, err)
+		block := latesthdr.Number.Uint64()
+		startBlocks[dest] = &block
+
+		// Send to the receiver on the destination chain paying with LINK token
+		var (
+			receiver = common.LeftPadBytes(state.Chains[dest].Receiver.Address().Bytes(), 32)
+			data     = []byte("hello world")
+			feeToken = linkToken.Address()
+		)
+
+		ccipMessage := router.ClientEVM2AnyMessage{
+			Receiver:     receiver,
+			Data:         data,
+			TokenAmounts: tokens[src],
+			FeeToken:     feeToken,
+			ExtraArgs:    nil,
+		}
+
+		// Get the fee Token Balance Before
+		feeTokenBalanceBefore, err := linkToken.BalanceOf(nil, e.Chains[src].DeployerKey.From)
+		require.NoError(t, err)
+
+		// Check the fee Amount
+		srcFee, err := state.Chains[src].Router.GetFee(nil, dest, ccipMessage)
+		require.NoError(t, err)
+
+		msgSentEvent := changeset.TestSendRequest(t, e, state, src, dest, false, ccipMessage)
+
+		expectedSeqNum[changeset.SourceDestPair{
+			SourceChainSelector: src,
+			DestChainSelector:   dest,
+		}] = msgSentEvent.SequenceNumber
+		expectedSeqNumExec[changeset.SourceDestPair{
+			SourceChainSelector: src,
+			DestChainSelector:   dest,
+		}] = []uint64{msgSentEvent.SequenceNumber}
+
+		// Check the fee token balance after the request and ensure fee tokens were spent
+		feeTokenBalanceAfter, err := linkToken.BalanceOf(nil, e.Chains[tenv.HomeChainSel].DeployerKey.From)
+		require.NoError(t, err)
+		require.Equal(t, feeTokenBalanceAfter, new(big.Int).Sub(feeTokenBalanceBefore, srcFee))
+
+		// Wait for all commit reports to land.
+		changeset.ConfirmCommitForAllWithExpectedSeqNums(t, e, state, expectedSeqNum, startBlocks)
+
+		// After commit is reported on all chains, token prices should be updated in FeeQuoter.
+		linkAddress := state.Chains[dest].LinkToken.Address()
+		feeQuoter := state.Chains[dest].FeeQuoter
+		timestampedPrice, err := feeQuoter.GetTokenPrice(nil, linkAddress)
+		require.NoError(t, err)
+		require.Equal(t, changeset.MockLinkPrice, timestampedPrice.Value)
+
+		// Wait for all exec reports to land
+		changeset.ConfirmExecWithSeqNrsForAll(t, e, state, expectedSeqNumExec, startBlocks)
+
+		balance, err := dstToken.BalanceOf(nil, state.Chains[tenv.FeedChainSel].Receiver.Address())
+		require.NoError(t, err)
+		require.Equal(t, twoCoins, balance)
+
+		// Delete from the expcted seq num the chain that was just tested so that we don't pass it to the
+		// commit report in the next test
+		delete(expectedSeqNum, sourceDestPair)
+	})
+
+	t.Run("Send Token Pay with native remote chain -> home", func(t *testing.T) {
```

Comment:
If the difference in this test is only about fee token, then we can pass that as an argument and simplify the test.